jenkins/pipelines/ci/tikv/tikv_ghpr_coverage.groovy

```diff
@@ -0,0 +1,117 @@
+    podTemplate(name: label, label: label,
+        nodeSelector: 'role_type=slave', instanceCap: 7,
+        workspaceVolume: emptyDirWorkspaceVolume(memory: true),
+        containers: [
+            containerTemplate(name: 'rust', image: "hub.pingcap.net/jenkins/tikv-cached-${pod_image_param}:latest",
+                alwaysPullImage: true, privileged: true,
+                resourceRequestCpu: '2', resourceRequestMemory: '8Gi',
+                ttyEnabled: true, command: 'cat'),
+        ]) {
+        node(label) {
+            println "[Debug Info] Debug command: kubectl -n jenkins-ci exec -ti ${NODE_NAME} bash"
+
+            container("rust") {
+                sh label: 'Prepare workspace', script: """
+                    cd \$HOME/tikv-src
+                    git fetch origin
+                    git checkout -f origin/${ghprbTargetBranch}
+                    rustup component add llvm-tools-preview
+                    cargo install grcov
+                """
+
+                def should_skip = sh (label: 'Check if can skip', script: """
+                cd \$HOME/tikv-src
+                if git log -1 | grep '\\[ci skip\\]'; then
+                    exit 0
+                fi
+                ghprbActualCommit=`git rev-parse HEAD`
+                curl --output /dev/null --silent --head --fail ${FILE_SERVER_URL}/download/coverage_check/${JOB_NAME}/\$ghprbActualCommit
+                """, returnStatus: true)
+                if (should_skip == 0) {
+                    throw new RuntimeException("ci skip")
+                }
+
+                sh label: 'Run test artifact', script: """
+                cd \$HOME/tikv-src
+                export FAIL_POINT=1
+                export ROCKSDB_SYS_SSE=1
+                export CARGO_INCREMENTAL=0
+                export RUSTFLAGS="-Zinstrument-coverage"
+
+                grpcio_ver=`grep -A 1 'name = "grpcio"' Cargo.lock | tail -n 1 | cut -d '"' -f 2`
+                if [[ ! "0.8.0" > "\$grpcio_ver" ]]; then
+                    echo using gcc 8
+                    source /opt/rh/devtoolset-8/enable
+                fi
+
+                env RUSTFLAGS="-Zinstrument-coverage" LLVM_PROFILE_FILE="tikv-%p-%m.profraw" FAIL_POINT=1 RUST_TEST_THREADS=1 EXTRA_CARGO_ARGS=--no-fail-fast make test || true
+                """
+
+                sh label: 'Post-build: Generating coverage', script: """
+                cd \$HOME/tikv-src
+                grcov . --binary-path ./target/debug/ -s . -t lcov --branch --ignore-not-existing --ignore "/*" --ignore "target/debug/*" -o lcov.info
+                """
+
+                withCredentials([string(credentialsId: 'TIKV_CODECOV_TOKEN', variable: 'CODECOV_TOKEN')]) {
+                    timeout(180) {
+                        sh label: 'Post-build: Uploading coverage', script: """
+                        cd \$HOME/tikv-src
+                        curl -OLs https://uploader.codecov.io/latest/linux/codecov
```

Comment:
codecov binary could be download from fileserver:
`curl -LO ${FILE_SERVER_URL}/download/cicd/ci-tools/codecov`